Add new System Logs Viewer (`logviewer`) module

iliajie opened the pull request:
This module will allow users to view all log files, in a familiar way to what `syslog` module did. The `systemd-journald` service is supported.

From this point it can grow more on its own.

Features:

  1. Keep to display all other modules' log files
  2. Keep ability to add custom logs
  3. Keep ACLs support
  4. Add better `journalctl` support
  5. Add a message, when `syslog` or `syslog-ng` is not installed, to tell a user to try a new module, instead of installing `rsyslog` package

**Examples:**

<img width="1099" alt="image" src="https://user-images.githubusercontent.com/4426533/170893324-43ae7396-12a0-416f-9d15-93279d2ed3df.png">
<img width="1154" alt="image" src="https://user-images.githubusercontent.com/4426533/170893356-7f4c8a10-358f-4c92-be60-44ece92375d9.png">

<img width="1097" alt="image" src="https://user-images.githubusercontent.com/4426533/170893287-a021b220-5c72-4276-98fd-5ad8dbb8bbfd.png">
<img width="1097" alt="image" src="https://user-images.githubusercontent.com/4426533/170893306-24071033-4eb1-4231-ab99-f785fc7b79c5.png">

jcameron commented:
Nice! Will this discover logs belonging to `syslog` though?

jcameron commented on logviewer/logviewer-lib.pl:
```diff
@@ -0,0 +1,133 @@
+{
+my $lines = $config{'lines'} || 1000;
+return !&has_command('journalctl') ? () : (
+	{ 'cmd' => "journalctl --lines $lines -p 0..3",
```
What do the numbers 0 to 3 mean?

swelljoe replied:
Priority.

```
-p, --priority=
           Filter output by message priorities or priority ranges. Takes
           either a single numeric or textual log level (i.e. between
           0/"emerg" and 7/"debug"), or a range of numeric/text log
           levels in the form FROM..TO. The log levels are the usual
           syslog log levels as documented in [syslog(3)](https://www.man7.org/linux/man-pages/man3/syslog.3.html), i.e.
           "emerg" (0), "alert" (1), "crit" (2), "err" (3),
           "warning" (4), "notice" (5), "info" (6), "debug" (7). If a
           single log level is specified, all messages with this log
           level or a lower (hence more important) log level are shown.
           If a range is specified, all messages within the range are
           shown, including both the start and the end value of the
           range. This will add "PRIORITY=" matches for the specified
           priorities.
```

I believe the text version would be more self-documenting. e.g. `emerg` instead of `0`.

iliajie commented:
> Nice! Will this discover logs belonging to `syslog` though?

Yes, it will, logs from Webmin, Apache and etc. Do you mean these ones?

> What do the numbers 0 to 3 mean?

That means (as described on the first page) in "Messages selected" column, to display messages that are error, critical, alert and emergency.

> I believe the text version would be more self-documenting. e.g. emerg instead of 0.

Yes, this is what I did at first (locally) but then I couldn't sort it (easily) properly (based on severity level) and I got back to using numerical ranges. I think you're right. I will change it to text based range.

iliajie commented:
Alright, the commit above fixed it. I reworked how the messages are displayed and named. Fixed sorting and few other bugs.

**Example:**
<img width="1189" alt="image" src="https://user-images.githubusercontent.com/4426533/170976694-b317cd54-f558-4072-a108-9ab0efe722a3.png">

<img width="1189" alt="image" src="https://user-images.githubusercontent.com/4426533/170976575-734c875c-7b12-4f72-bd00-07e5ca749818.png">

iliajie commented:
> Nice! Will this discover logs belonging to syslog though?

Oh, I see what you mean now. I didn't think before that the new module would need to list any `syslog` files. I have fixed that on [this commit](https://github.com/webmin/webmin/pull/1655/commits/88c6e06c038831b3a118b7d88773c875c4d87bcc). Now, if `syslog` is installed and available it will also include the logs from it.

<img width="891" alt="image" src="https://user-images.githubusercontent.com/4426533/170996933-29dcf8a1-8bb5-4005-b792-586c0533fc26.png">

jcameron commented:
Excellent! Can you also do the same for logs managed by the `syslog-ng` module?

iliajie commented:
> Excellent! 

Thanks!

> Can you also do the same for logs managed by the `syslog-ng` module?

Well, yeah, if you think it should be considered too.

jcameron commented:
We certainly still have users running `syslog-ng` .. and it should be pretty trivial to add code to fetch the list of log files from the Syslog-NG config if the module is installed.

iliajie commented:
Can do, tomorrow.

iliajie commented:
Where are we at here?

jcameron commented:
I think we can call this done, and ready for inclusion in the next release!

Also, maybe in a future PR you could link to this module from the syslog and syslog-ng modules for viewing logs?

iliajie commented:
It's already linked from `syslog` and `syslog-ng`. I mean, I already added it.

Is this what you mean?
https://github.com/webmin/webmin/pull/1655/commits/3a7dda67cd9425a9b08308ed906d8c57fac6a77d

jcameron commented:
I meant the other way - if the user clicks on a link to view logs on the `syslog` module, it could just link to this new one

jcameron commented:
I meant the other way - if the user clicks on a link to view logs on the `syslog` module, it could just link to this new one

iliajie commented:
Well, this is what my patch does? It mentions a new `logview` module but only if `syslog` or `syslog-ng` not installed.

Like that -

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/4426533/172011852-689df27b-1547-498c-bd63-65ffaaabedf8.png">


Is this what you mean?

jcameron commented:
No, I meant that if the Syslog is installed, in the `syslog` module we could link to the `logview` module to view logs.

iliajie commented:
Like FirewallD message in Firewall? Like yellow alert on the top?

jcameron commented:
It might be easier for me to just make the change myself so you can see what I mean :-)
